src/Headless/OpenGLContext/OpenGLContext.hpp

```diff
@@ -0,0 +1,145 @@
+#pragma once
+#include <Core/Utils/Observable.hpp>
+#include <Headless/RaHeadless.hpp>
+
+#include <array>
+#include <functional>
+#include <string>
+struct GLFWwindow;
+
+namespace Ra {
+namespace Headless {
+
+/**
+ * This class defines GLFW based OpenGL Context.
+ * There are two use cases of such a context :
+ *   - headless rendering application.
+ *   - simple window with limited interaction
```

Comment:
Do we really need to write these use case ?
I mean: if new use case are added, we will have to update the list, which we will likely forget to do.

Reply:
I think we do. This class is, in fact, a wrapper to glfw OpenGL context and window management but exposes only a few of glfw interaction capabilities. 
So either we use it for headless rendering (without user interaction) or we use it with simple interaction based on simplified keyboard, mouse, scroll and resize event wrt glfw capabilities. 
And even if one can develop quite complex interactive application with this OpenGLContext by using an external GUI system, e.g.,  imGUI, the interaction capabilities will be limited by the wrapping we do. 
If we change this wrapping in the future, we'll change also the doc and if a developper forget to do this, PR reviewers will tell them to add doc ....

Reply:
ok, it was not clear to me if these use cases were the only ones possible, or if other variants could be implemented.